lib/AST/ASTImporter.cpp

```diff
@@ -6806,31 +6804,32 @@ Decl *ASTImporter::GetVAListTag(Decl *FromD) {
 }
 
 Decl *ASTImporter::Import(Decl *FromD) {
+
   if (!FromD)
     return nullptr;
 
+  if (VisitedDecls.count(FromD) > 0)
```

Comment:
Does the fix only work with VisitedDecls? If a Decl is visited it is not guaranteed that it is already finished with the import right? So we might return `nullptr` here in some cases which might cause failures. Do I miss something?

Reply:
I think I answered all these questions in the summary of this PR.
Could you please read the summary of this PR?

Reply:
The `VisitedDecls` is only needed to handle the templated decl vs described template circle.

Reply:
Oh, I see. Thanks. I am really uncomfortable about changing the meaning of the return value of Import call because the rest of the importer was built with a different assumption. I am afraid of this change causing lots of regressions.

Reply:
But the null return value indicated only error until now? We will have only more cases with non-null return value. (If null is returned in the new case and non-null in the old this should mean there is a recursive-import related problem.)

Reply:
Okay, I think it may be the best to keep the original semantics of the return value: ie when nullptr there was an error.
I will add a new member to the importer with which we can use to query if we have encountered an already visited node during the import. This is ugly, but at least we could distinguish the error from the already visited state properly.